core/src/eval/operation.rs

```diff
@@ -1571,15 +1567,14 @@ impl<R: ImportResolver, C: Cache> VirtualMachine<R, C> {
                                             call_stack: std::mem::take(&mut self.call_stack),
                                         })
                                     }
-                                    _ => Err(EvalError::FieldMissing(
-                                        id.into_inner(),
-                                        String::from("(.$)"),
-                                        RichTerm {
-                                            term: t2,
-                                            pos: pos2,
-                                        },
+                                    _ => Err(EvalError::FieldMissing {
+                                        id: ident,
+                                        field_names: record
+                                            .field_names(RecordOpKind::IgnoreEmptyOpt),
+                                        operator: String::from("(.$)"),
```

Comment:
Not related to this PR, but where is that `.$` coming from?

Reply:
Oh, good catch. In fact the operator wasn't shown before this PR (basically the `notes` were created but never used - it's unrelated to this PR, it's just that I needed to add the suggestion to the notes, so I realized, and it was in fact surprising that the rust compiler didn't yell at me for unused value). So we might have let old name slipping through. But I believe this was the original syntax for dynamic access field, like `foo.$x` is the current `foo."%{x}"`, because we used to have `$` as the interpolation sign and we didn't have quoted fields yet, such as `foo."bar^baz"`. When we added the later syntax, it felt better to avoid keeping the strange `.$` which could just be subsumed by quoted field access + interpolation.